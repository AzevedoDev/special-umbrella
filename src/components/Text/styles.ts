import styled, { css } from 'styled-components'
import { TextProps } from '.'

export const TextComponent = styled.p<TextProps>`
  ${({ theme, color }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${color && theme.colors[color!]};
    font-weight: ${theme.font.normal};
    text-align: center;
    font-family: ${theme.font.family};
  `}
`
