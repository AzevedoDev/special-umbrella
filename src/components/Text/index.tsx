import * as S from './styles'

export type TextProps = {
  children: React.ReactNode
  color?: 'white' | 'black'
}

const Text = ({ children, color = 'white' }: TextProps) => (
  <S.TextComponent color={color}>{children}</S.TextComponent>
)

export default Text
