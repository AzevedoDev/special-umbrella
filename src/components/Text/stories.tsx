import { Story, Meta } from '@storybook/react/types-6-0'
import Text, { TextProps } from '.'

export default {
  title: 'Text',
  component: Text,
  args: {
    children: 'Texto'
  },
  argTypes: {
    color: {
      control: {
        type: 'select',
        options: ['white', 'black']
      }
    }
  }
} as Meta

export const Default: Story<TextProps> = (args) => <Text {...args} />
