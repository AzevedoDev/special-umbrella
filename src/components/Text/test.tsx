import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Text from '.'

describe('<Text />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Text color="white">Text</Text>)

    expect(screen.getByText(/Text/)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
