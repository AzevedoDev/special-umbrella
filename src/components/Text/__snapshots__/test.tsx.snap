// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render the heading 1`] = `
.c0 {
  font-size: 1.4rem;
  color: #FFFFFF;
  font-weight: 500;
  text-align: center;
  font-family: 'Montserrat',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

<p
  class="c0"
  color="white"
>
  Text
</p>
`
